### PR TITLE
[KIALI-153] Show Graph legend

### PR DESCRIPTION
**Pull request by aljesusg**
legend

Still working, please feedback ^^ @abonas @serenamarie125 @xeviknal

### CODE REVIEW

**Comment by aljesusg** [comment]
@serenamarie125 related with colours, Is there any place where patternfly colors are defined?

**Comment by abonas** [comment]
I think that the "show legend" should not be hidden in the "..." part, most of the users won't understand it is there. There is enough room for "hide/show legend" imo in the toolbar.

**Comment by abonas** [comment]
and yes, the colors should match PF and should match the graph itself :)

**Comment by aljesusg** [comment]
@abonas I discussed this with @xeviknal , the proposal of serena was this, I don't know why the button is inside the dropdownkebap, I'll change
legend

About colors ....yes ^^ I need to change them. Thanks

**Comment by abonas** [comment]
@serenamarie125 your feedback is appreciated :)

**Comment by aljesusg** [comment]
Updated Legend, changed to button and PF Colors:
```javascript
const PFCOLORS = {
  'pf-red-100': '#cc0000',
  'pf-orange-300': '#f39d3c',
  'pf-blue': '#0088ce',
  'pf-black': '#030303'
};
```
legend

**Comment by serenamarie125** [comment]
@aljesusg, pf-orange-400 (#ec7a08) pf-red-100 (#cc0000) pf-green-400 (#3f9c35).

**Comment by serenamarie125** [comment]
I think that the legend should show up on the bottom left by default.
The items in the legend should have arrows, just show the line, and please make the line thickness thinner.

**Comment by aljesusg** [comment]
Ok, I saw that in the graph they are using the base colors. Should I change the colors of graph too?

**Comment by serenamarie125** [comment]
Also, that mock is old :) Here's the latest.  We changed the color from blue to green 

screen shot 2018-05-24 at 8 33 07 am

**Comment by abonas** [comment]
> The items in the legend should have arrows, just show the line, and please make the line thickness thinner.

@serenamarie125 
'should have arrows' means should **not** have arrows, right? :)

**Comment by aljesusg** [comment]
Thanks @serenamarie125 
I changed the app colors to PF colors, about arrows? ^^

**Comment by serenamarie125** [comment]
sorry bout that @aljesusg !  Missed a word
The items in the legend should *not* have arrows, just show the line, and please make the line thickness thinner.

**Comment by aljesusg** [comment]
legend

Last release @abonas & @serenamarie125

**Comment by abonas** [comment]
@aljesusg looks great!
please note @serenamarie125 comment about legend's default position: "the legend should show up on the bottom left by default."

**Comment by aljesusg** [comment]
legend

Last release, @abonas @serenamarie125 please review UI

File: src/pages/ServiceGraph/ServiceGraphPage.tsx [hunk 1]
```diff
@@ -37,7 +40,8 @@ export default class ServiceGraphPage extends React.Component<ServiceGraphPagePr
     super(props);
 
     this.state = {
-      summaryData: { summaryType: 'graph', summaryTarget: null }
+      summaryData: { summaryType: 'graph', summaryTarget: null },
+      hideLegend: true
```
Comment (mtho11) [inline src/pages/ServiceGraph/ServiceGraphPage.tsx]: Please put `hideLegend` into Redux store. The `state` for _ServiceGraphPage_ has actually already been removed and _ServiceGraphPage_ is a stateless component once the PR https://github.com/kiali/kiali-ui/pull/354 is merged.

File: src/pages/ServiceGraph/ServiceGraphPage.tsx [hunk 2]
```diff
@@ -78,6 +82,10 @@ export default class ServiceGraphPage extends React.Component<ServiceGraphPagePr
     this.loadGraphDataFromBackend();
   };
 
+  handleLegendClick = () => {
+    this.setState({ hideLegend: !this.state.hideLegend });
```
Comment (mtho11) [inline src/pages/ServiceGraph/ServiceGraphPage.tsx]: Please dispatch a Redux Action instead

File: src/components/GraphFilter/GraphLegend.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      lineBase['backgroundColor'] = this.props.color;
+    }
+    this.state = {
+      line: lineBase
+    };
+  }
+
+  render() {
+    return (
+      <div style={{ marginTop: 20, clear: 'both' }}>
```
[the remark this answers is not in the document]
Comment (aljesusg) [reply src/components/GraphFilter/GraphLegend.tsx]: wowow i didn't know that you added Typestyle. perfect thanks

**Comment by aljesusg** [comment]
@mtho11 I did the changes

File: src/components/GraphFilter/GraphLegend.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+export interface GraphLegendState {}
+
+export interface ArrowProps {
+  color: string;
+  label: string;
+  dashed: boolean;
+}
+
+class Arrow extends React.Component<ArrowProps> {
```
[the remark this answers is not in the document]
Comment (aljesusg) [reply src/components/GraphFilter/GraphLegend.tsx]: Ok, I'll do this changes while I wait @serenamarie125 review thanks @mtho11

File: src/components/GraphFilter/GraphLegend.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  getLinebase = () => {
+    if (this.props.dashed) {
+      return style({
+        marginTop: '5px',
+        width: '90px',
+        height: '5px',
+        float: 'left',
+        backgroundImage:
+          'repeating-linear-gradient(90deg, ' +
```
[the remark this answers is not in the document]
Comment (aljesusg) [reply src/components/GraphFilter/GraphLegend.tsx]: Yes more cleaner now ^^

**Comment by aljesusg** [comment]
Thanks @mtho11

**Pull request merged**